src/api_config.cpp

```diff
@@ -92,6 +92,25 @@ void api_config::load_from_file(const std::string &filename) {
 			}
 		}
 
+		// read the [log] settings
+		int log_level = pt.get("log.level", (int)loguru::Verbosity_INFO);
+		if (log_level < -3 || log_level > 9)
+			throw std::runtime_error("Invalid log.level (valid range: -3 to 9");
+
+		std::string log_file = pt.get("log.file", "");
+		if (!log_file.empty()) {
+			loguru::add_file(log_file.c_str(), loguru::Append, log_level);
+			// don't duplicate log to stderr
+			loguru::g_stderr_verbosity = -9;
+		} else
+			loguru::g_stderr_verbosity = log_level;
+
+		// log config filename only after setting the verbosity level
```

Comment:
This log statement should remain at the end. We should only see "Configuration loaded ..." after we've finished parsing the loaded config.